### ISSUE TITLE
Question: how to tell "plot_genes_branched_heatmap" the right pre-branch and branch cells?

### ISSUE THREAD

**Issue by qrzhang** [cole-trapnell-lab/monocle-release#111]
Hi, 

I want to visualize changes for all the genes significantly branch dependent using `plot_genes_branched_heatmap()`.  The simplified trajectory of my data is as following.

![monocle_question](https://user-images.githubusercontent.com/19705471/35688671-a7df641c-07ac-11e8-8f79-93bdc4988231.png)

Firstly, I provide a branch point. The code is `plot_genes_branched_heatmap(data[genes,] , branch_point = 1, num_clusters = 6)`. When looked at the expression of marker genes in the heatmap,  I found that only the beginning part of all pre-branch cells (State 1 cells) are annotated as pre-branch. 

After reading the source code of this function, I realized that, by default, only cells have same state with  that Pseudotime == 0 cell are regarded as `common_ancestor_cells`. Therefore, I provide the states of cells in two branches and the code is now `plot_genes_branched_heatmap(data[genes,] ,branch_point = 1, num_clusters = 6, branch_states = c(4:7))`. However, this time an error message shows up, 
> Error in as.igraph.vs(graph, from) : Invalid vertex names
>Calls: plot_genes_branched_heatmap ... buildBranchCellDataSet -> shortest_paths -> as.igraph.vs

Does this mean that multiple branch_states values are not supported? 

Till now, I have no idea about how to tell `plot_genes_branched_heatmap` the right pre-branch cells(State 1, 2, 3) and branch cells(State 4, 5, 6, 7)? Will appreciate any help :)

**Comment by SmileZero** [comment]
> #' @param branch_states The two states to compare in the heatmap. Mutually exclusive with branch_point. 
I don't think you can use the branch_states and branch_point at the same time. Also, this function can only compare two states, and there is another function can compare multiple states.

For your question, if you choose the branch point 2, then pre-branch is State 1 since its pseudo time is the least.

**Comment by Xiaojieqiu** [comment]
Thanks @SmileZero and @qrzhang for the nice suggestions and question. As @SmileZero  said, we cannot compare more than two states with  `plot_genes_branched_heatmap` but the `plot_multiple_branches_heatmap` can allow us to visualize multiple branches. 

regarding to only `state 1` is annotated as `pre-branch`, that is potentially a bug in our code. We will fix it in our next release. thanks for pointing this out

**Comment by BrianLohman** [comment]
Similarly, how do you tell which "cell fate" corresponds to which cell state in the resulting heatmap? The default labels are "Cell Fate 1" and "Cell Fate 2". For example, here where we are talking about branch point 2, does cell fate 1 correspond to state 2 or state 3?

**Comment by EDePasquale** [comment]
I have the same question as BrianLohman. I had to figure it out with gene enrichment analysis, and thankfully the 2 fates were quite distinct, but I feel there should be some way to tell which is on the left and which on the right.

**Comment by Emma-China** [comment]
> I have the same question as BrianLohman. I had to figure it out with gene enrichment analysis, and thankfully the 2 fates were quite distinct, but I feel there should be some way to tell which is on the left and which on the right.
@EDePasquale How do you do gene enrichment analysis?Would you share me your code?thanks!

**Comment by MubasherMohammed** [comment]
I still can’t find answers for how to map cell fate in Pseudotime branch heatmap with cell state in the trajectory.  Does any one managed? Regardless performing gene enrichment
